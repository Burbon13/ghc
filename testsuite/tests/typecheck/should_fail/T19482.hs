{-# LANGUAGE DataKinds #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE PolyKinds #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE AllowAmbiguousTypes #-}

module T19482 where

testF :: forall a (b :: [a]). ()
testF = ()

-- Both Test1 and Test2 led to "no skolem info" errors in GHC 9.0

-- Test1
class BugClass k where
    bugList :: ()
instance BugClass ((s : sx) :: [r]) where
    bugList = testF @r @s

-- Test2
foo :: forall r (s::r). ()
foo = testF @r @s
