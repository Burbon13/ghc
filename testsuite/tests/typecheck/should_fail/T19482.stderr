
T19482.hs:19:25: error:
    • Expected kind ‘[r]’, but ‘s’ has kind ‘r’
      ‘r’ is a rigid type variable bound by
        the instance declaration
        at T19482.hs:18:10-35
    • In the type ‘s’
      In the expression: testF @r @s
      In an equation for ‘bugList’: bugList = testF @r @s

T19482.hs:23:17: error:
    • Expected kind ‘[r]’, but ‘s’ has kind ‘r’
      ‘r’ is a rigid type variable bound by
        the type signature for:
          foo :: forall r (s :: r). ()
        at T19482.hs:22:1-26
    • In the type ‘s’
      In the expression: testF @r @s
      In an equation for ‘foo’: foo = testF @r @s
